homeassistant/components/fido/sensor.py

```diff
@@ -97,32 +182,18 @@ def __init__(self, fido_data, sensor_type, name, number):
         self.client_name = name
         self._number = number
         self.type = sensor_type
-        self._name = SENSOR_TYPES[sensor_type][0]
-        self._unit_of_measurement = SENSOR_TYPES[sensor_type][1]
-        self._icon = SENSOR_TYPES[sensor_type][2]
+        metadata = SENSOR_TYPES[sensor_type]
```

Comment:
If you pass in the entity description directly into the constructor, you can do:

```suggestion
        self.entity_description = entity_description
```

Which will take care of some of the other attributes automatically

Reply:
Done.